snykTask/src/task-lib.ts

```diff
@@ -157,7 +157,12 @@ export function doVulnerabilitiesExistForFailureThreshold(
   const json = JSON.parse(file);
   const thresholdOrdinal = getSeverityOrdinal(threshold);
 
-  if (Array.isArray(json)) {
+  // code test json identified by $schema property and does not describe issues severity
+  if (json['$schema'] && json['runs'][0]['results'].length > 0) {
```

Comment:
suggestion: it might be worth investigating the output payload for a "severity level". On initial inspection, it looks as though an individual `result` has property `level` which can be mapped and compared against the `thresholdOrdinal`:

```json
{
  "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
  "version": "2.1.0",
  "runs": [
    {
      "tool": {...},
      "results": [
        {
          "ruleId": "javascript/OR",
          "ruleIndex": 0,
          "level": "warning", // can be "note" | "warning" | "error"
          ...
        }
      ],
      "properties": {...}
    }
  ]
}
```

Reply:
ok, let me confirm with ask-code team on this sarif schema. Looks promising. from the schema
```
        "level": {
          "description": "A value specifying the severity level of the notification.",
          "default": "warning",
          "enum": [ "none", "note", "warning", "error" ],
          "type": "string"
        },
```

Reply:
refactored with :-
-  a `failOnThreshold` condition matching or higher severity than `codeSeverityThreshold` based on this mapping of snyk code output json issue `level` to severity at `snyk-to-html`.
- test-cases of task args checking this _matching or higher severity_ i.e. `>=` condition.
- test of snyk code output json to a `isSnykCodeOutput` function call.